File: FSUIKit/Classes/FSDBJeView.m
//
//  FSDBJeView.m
//  myhome
//
//  Created by Fudongdong on 2017/12/21.
//  Copyright © 2017年 fuhope. All rights reserved.
//

#import "FSDBJeView.h"

@implementation FSDBJeView

- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        [self jeDesignViews];
    }
    return self;
}

- (void)jeDesignViews{
    self.backgroundColor = [UIColor whiteColor];
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapAction)];
    [self addGestureRecognizer:tap];
    
    CGSize size = self.frame.size;
    for (int x = 0; x < 2; x ++) {
        FSLabelTextField *textField = [[FSLabelTextField alloc] initWithFrame:CGRectMake(0, 10 + 45 * x, size.width, 44) text:x?NSLocalizedString(@"Note", nil):NSLocalizedString(@"Money", nil) textFieldText:nil placeholder:x?NSLocalizedString(@"Please input message", nil):NSLocalizedString(@"Please input money", nil)];
        textField.textField.keyboardType = x? UIKeyboardTypeDefault:UIKeyboardTypeDecimalPad;
        [self addSubview:textField];
        textField.label.font = [UIFont systemFontOfSize:17];
        textField.label.textColor = [UIColor blackColor];
        
        UIView *line = [[UIView alloc] initWithFrame:CGRectMake(15, textField.frame.origin.y + textField.frame.size.height, size.width - 15, .6)];
        line.backgroundColor = [UIColor colorWithRed:200 / 255.0 green:200 / 255.0 blue:200 / 255.0 alpha:1.0];
        [self addSubview:line];
        if (x) {
            self.bzTF = textField;
        }else{
            self.jeTF = textField;
        }
    }
}

- (void)tapAction{
    if (self.tapEvent) {
        self.tapEvent(self);
    }
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: FSUIKit/Classes/Swift/FSDBJeView.swift
// FSDBJeViewS.swift
// Translated from FSDBJeView.h/m

import UIKit

open class FSDBJeView: UIView {
    
    public lazy var jeTF: FSLabelTextField = {
        let textField = FSLabelTextField(
            frame: CGRect(x: 0, y: 10, width: frame.size.width, height: 44),
            text: "金额",
            textFieldText: nil,
            placeholder: "请输入金额"
        )
        
        textField.textField.keyboardType = .decimalPad
        addSubview(textField)
        textField.label.font = UIFont.systemFont(ofSize: 17)
        textField.label.textColor = .black
        
        let line = UIView(frame: CGRect(x: 15, y: textField.frame.origin.y + textField.frame.height, width: frame.size.width - 15, height: 0.6))
        line.backgroundColor = UIColor(red: 200/255.0, green: 200/255.0, blue: 200/255.0, alpha: 1.0)
        addSubview(line)

        return textField
    }()
    
    public lazy var bzTF: FSLabelTextField = {
        let textField = FSLabelTextField(
            frame: CGRect(x: 0, y: 10 + 45, width: frame.size.width, height: 44),
            text: "备注",
            textFieldText: nil,
            placeholder: "输入备注"
        )
        textField.textField.keyboardType = .default
        addSubview(textField)
        textField.label.font = UIFont.systemFont(ofSize: 17)
        textField.label.textColor = .black
        
        let line = UIView(frame: CGRect(x: 15, y: textField.frame.origin.y + textField.frame.height, width: frame.size.width - 15, height: 0.6))
        line.backgroundColor = UIColor(red: 200/255.0, green: 200/255.0, blue: 200/255.0, alpha: 1.0)
        addSubview(line)

        return textField
    }()
    
    public var tapEvent: ((FSDBJeView) -> Void)?
    
    public override init(frame: CGRect) {
        super.init(frame: frame)
        jeDesignViews()
    }
    
    required public init?(coder: NSCoder) {
        super.init(coder: coder)
        jeDesignViews()
    }
    
    private func jeDesignViews() {
        backgroundColor = .white
        
        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(tapAction))
        addGestureRecognizer(tapGesture)
        
        _ = jeTF
        _ = bzTF
    }
    
    @objc private func tapAction() {
        tapEvent?(self)
    }
}
